src/network/tags.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-FileCopyrightText: 2021 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import produce from 'immer';
+import {
+	CreateTagRequest,
+	CreateTagResponse,
+	TagActionRequest,
+	TagActionResponse
+} from '../../types';
+import { Tag } from '../../types/tags';
+import { SHELL_APP_ID } from '../constants';
+import { useTagStore } from '../store/tags';
+import { getSoapFetch } from './fetch';
+
+const set = useTagStore.setState;
+export const createTag = (tag: Omit<Tag, 'id'>): Promise<CreateTagResponse> =>
+	getSoapFetch(SHELL_APP_ID)<CreateTagRequest, CreateTagResponse>('CreateTag', {
+		_jsns: 'urn:zimbraMail',
+		tag
+	});
```

Comment:
Just a question ,if we need to rename and changeTagColor at the same time ,then do we have to use 2 separate calls renameTag & changeTagColor ? Or can't we create 'updated' action also to modify multiple properties at the same time?

Reply:
it makes sense, considering the possibility of an 'edit tag' modal